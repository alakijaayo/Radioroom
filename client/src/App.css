.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* client/src/App.css */
div,
button {
  margin: 10px;
}

.spotify-login-btn {
  background: #84bd00;
  border-radius: 400px;
  height: 48px;
  margin: 8px;
  padding: 13px 17px;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  display: inline-block;
  color: #fff;
  text-align: center;
  transition: all 0.3s;
  white-space: nowrap;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  user-select: none;
  border: 1px solid transparent;
  text-decoration: none;
  box-sizing: border-box;
}
