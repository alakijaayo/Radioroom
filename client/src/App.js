import React, { Component } from 'react';
import './App.css';
import SpotifyWebApi from 'spotify-web-api-js';
import User from './User.js';
import NowPlaying from './NowPlaying.js';
import Queue from './Queue.js';
import Search from './Search.js';
import Player from './Player.js';
import Chat from './Chat.js';
import io from 'socket.io-client';

const spotifyApi = new SpotifyWebApi();
const socket = io(
  process.env.NODE_ENV === 'production'
    ? 'https://radioroomserver.herokuapp.com'
    : 'http://localhost:8888'
);

class App extends Component {
  constructor() {
    super();
    const params = this.getHashParams();
    socket.on(
      'Play Track',
      function(track) {
        this.player.playTrack(track.uri);
        this.setState({
          nowPlaying: {
            artist: track.artist,
            track: track.track,
            albumArt: track.artwork
          }
        });
      }.bind(this)
    );
    socket.on(
      'Queue Updated',
      function(queue) {
        console.log(queue);
        this.setState({
          upNext: queue
        });
      }.bind(this)
    );
    socket.on(
      'Chat Updated',
      function(messages) {
        console.log(messages);
        this.setState({
          chat: messages
        });
      }.bind(this)
    );
    this.token = params.access_token;
    this.refreshToken = params.refresh_token;
    this.timerId = 0;
    this.state = {
      loggedIn: this.token ? true : false,
      nowPlaying: { artist: '', track: '', albumArt: '' },
      user: {
        name: params.user_name,
        id: params.user_id,
        imageUrl: params.user_image_url
      }
    };
    this.addToPlaylist = this.addToPlaylist.bind(this);
    this.checkPlayerReady = this.checkPlayerReady.bind(this);
    this.addChatMessage = this.addChatMessage.bind(this);
    this.vote = this.vote.bind(this);
  }

  componentDidMount() {
    if (this.token) {
      spotifyApi.setAccessToken(this.token);
      if (window.PlayerReady) {
        this.player = new Player(this.token);
      } else {
        this.timerId = setInterval(this.checkPlayerReady, 100);
      }
    }
  }

  getHashParams() {
    var hashParams = {};
    var e,
      r = /([^&;=]+)=?([^&;]*)/g,
      q = window.location.hash.substring(1);
    e = r.exec(q);
    while (e) {
      hashParams[e[1]] = decodeURIComponent(e[2]);
      e = r.exec(q);
    }
    return hashParams;
  }

  getNowPlaying() {
    spotifyApi.getMyCurrentPlaybackState().then(response => {
      this.setState({
        nowPlaying: {
          artist: response.item.artists[0].name,
          track: response.item.name,
          albumArt: response.item.album.images[0].url
        }
      });
    });
  }

  addToPlaylist(spotifyTrack) {
    let queuedTrack = {
      uri: spotifyTrack.uri,
      artist: spotifyTrack.artists[0].name,
      track: spotifyTrack.name,
      artwork: spotifyTrack.album.images[0].url,
      duration: spotifyTrack.duration_ms
    };
    socket.emit('add to queue', JSON.stringify(queuedTrack));
  }

  checkPlayerReady() {
    if (window.PlayerReady) {
      this.player = new Player(this.token);
      clearInterval(this.timerId);
    }
  }

  vote(uri, vote) {
    const msg = vote === 1 ? 'vote up' : 'vote down';
    socket.emit(msg, uri);
  }

  addChatMessage(msg) {
    let chatMsg = {
      text: msg,
      user_url: this.state.user.imageUrl
    };
    socket.emit('chat message', JSON.stringify(chatMsg));
  }

  render() {
    let host =
      process.env.NODE_ENV === 'production'
        ? 'https://radioroomserver.herokuapp.com'
        : 'http://localhost:8888';
    return (
      <div className="App">
        <h1>RadioRoom</h1>
        {this.state.loggedIn ? (
          <div>
            <User user={this.state.user} />
            <NowPlaying nowPlaying={this.state.nowPlaying} />
            <Queue tracks={this.state.upNext} vote={this.vote} />
            <Search
              spotifyApi={spotifyApi}
              addToPlaylist={this.addToPlaylist}
            />
            <Chat
              messages={this.state.chat}
              addChatMessage={this.addChatMessage}
            />
          </div>
        ) : (
          <div>
            <h2>To get started, please login via Spotify</h2>
            <p>
              <strong>Note:</strong> To enjoy the full experience of RadioRoom
              you need to have a Spotify Premium subscription
            </p>
            <a className="spotify-login-btn" href={host}>
              Login to Spotify
            </a>
          </div>
        )}
      </div>
    );
  }
}

export default App;
