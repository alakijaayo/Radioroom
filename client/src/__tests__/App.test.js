import React from 'react';
import ReactDom from 'react-dom';
import App from '../App';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<App />, div);
  ReactDom.unmountComponentAtNode(div);
});

describe('App component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has a login text', () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find('a').text();
    expect(text).toEqual('Login to Spotify');
  });

  it('renders link to Spotify', () => {
    const wrapper = shallow(<App />);
    const linkToSpotify = wrapper.find('a');
    expect(linkToSpotify.length).toBe(1);
  });

  // it('shows a track in the "Now Playing" section', () => {
  //   const wrapper = shallow(<App />);
  //   const title = wrapper.find('.songTitle').text();
  //   expect(title).toContain('Now Playing: ');
  // });
});
