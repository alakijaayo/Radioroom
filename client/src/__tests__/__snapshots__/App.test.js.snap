// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="d-flex w-100"
  >
    <div
      className="align-self-center"
    >
      <img
        alt="user profile"
        src="/favicons/favicon-96x96.png"
      />
    </div>
    <div
      className="align-self-center flex-grow-1"
    >
      <h1
        className="text-left"
      >
        RadioRoom
      </h1>
    </div>
    
    <div>
      <img
        alt="user profile"
        className="align-self-center rounded-circle"
        src="/img/user.png"
        style={
          Object {
            "height": 96,
          }
        }
      />
    </div>
  </div>
  <div>
    <h2>
      To get started, please login via Spotify
    </h2>
    <p>
      <strong>
        Note:
      </strong>
       To enjoy the full experience of RadioRoom you need to have a Spotify Premium subscription
    </p>
    <a
      className="spotify-login-btn"
      href="http://localhost:8888"
    >
      Login to Spotify
    </a>
  </div>
  <div
    className="fixed-bottom px-auto w-100"
  >
    <div
      aria-label="Navigation Chat, Home and Search"
      className="btn-group btn-group-lg"
      role="group"
    >
      <button
        className="btn btn-secondary m-0"
        data-toggle="button"
        onClick={[Function]}
        type="button"
      >
        Chat
      </button>
      <button
        className="btn btn-secondary active m-0"
        data-toggle="button"
        onClick={[Function]}
        type="button"
      >
        Now Playing
      </button>
      <button
        className="btn btn-secondary m-0"
        data-toggle="button"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
